crates/ide-diagnostics/src/lib.rs

```diff
@@ -172,9 +197,12 @@ pub fn diagnostics(
         }),
     );
 
-    for node in parse.tree().syntax().descendants() {
+    let parse = sema.parse(file_id);
```

Comment:
Let's replace the `let parse = db.parse(file_id);` above with this instead

Reply:
`db.parse` above also contains syntax errors, which is used later. Is it possible to get syntax errors from `sema.parse`?

Reply:
Ah we use the actial `Parse` there, I didn't see that, then this is fine